source/extensions/tracers/xray/tracer.cc

```diff
@@ -35,10 +33,9 @@ constexpr auto XRaySerializationVersion = "1";
 //
 // For more details see:
 // https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-fields
-std::string generateTraceId(SystemTime point_in_time) {
+std::string generateTraceId(SystemTime point_in_time, Random::RandomGenerator& random) {
```

Comment:
Can we just remove time entirely from this and just use the UUID? cc @lavignes 

/wait

Reply:
I believe according to https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-fields we need the request time for the second part of the identifier

Reply:
Yeah unfortunately it is a requirement that trace ids have the epoch time be embedded in them.

I'm not entirely sure how _required_ this is, but it is best to stick to the spec.

Reply:
OK sounds good. @lavignes an you review and then I can stamp? Thank you.